accounts/src/account_service.rs

```diff
@@ -0,0 +1,302 @@
+                        },
+                    )
+                    .await?
+                    .iter()
+                    .map(|(pk, _)| *pk)
+                    .collect_vec();
+                accounts.append(&mut rpc_acc);
+            }
+        }
+        log::info!("Fetching {} accounts", accounts.len());
+        for accounts in accounts.chunks(max_request_in_parallel * NB_ACCOUNTS_IN_GMA) {
+            for accounts in accounts.chunks(NB_ACCOUNTS_IN_GMA) {
+                let mut fetch_accounts = vec![];
+                let mut updated_slot = 0;
+                for _ in 0..NB_RETRY {
+                    let accounts = rpc_client
+                        .get_multiple_accounts_with_config(
+                            accounts,
+                            RpcAccountInfoConfig {
+                                encoding: Some(solana_account_decoder::UiAccountEncoding::Base64),
+                                data_slice: None,
+                                commitment: Some(CommitmentConfig::finalized()),
+                                min_context_slot: None,
+                            },
+                        )
+                        .await;
+                    match accounts {
+                        Ok(response) => {
+                            fetch_accounts = response.value;
+                            updated_slot = response.context.slot;
+                            break;
+                        }
+                        Err(e) => {
+                            // retry
+                            log::error!("Error fetching all the accounts {e:?}, retrying");
+                            continue;
+                        }
+                    }
+                }
+                for (index, account) in fetch_accounts.iter().enumerate() {
+                    if let Some(account) = account {
+                        self.account_store
+                            .initilize_account(AccountData {
+                                pubkey: accounts[index],
+                                account: account.clone(),
+                                updated_slot,
+                            })
+                            .await;
+                    }
+                }
+            }
+        }
+        log::info!("{} accounts successfully fetched", accounts.len());
+        Ok(())
+    }
+
+    pub fn process_account_stream(
+        &self,
+        mut account_stream: AccountStream,
+        mut block_stream: BlockStream,
+    ) -> Vec<AnyhowJoinHandle> {
+        let this = self.clone();
+        let processed_task = tokio::spawn(async move {
+            loop {
+                match account_stream.recv().await {
+                    Ok(account_notification) => {
+                        this.account_store
+                            .update_account(
+                                account_notification.data,
+                                account_notification.commitment,
```

Comment:
did you consider adding the commitment to AccountData; the updated_slot and the respective commitment level should be packed together - no?

Reply:
Commitment cannot be part of account data and should remain separate from account data..